### PR TITLE
add rebuild/ca-rebuild-aba

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
ca-rebuild-aba

### CODE REVIEW

File: src/pages/ca-rebuild-aba/form/CARebuildForm.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                displayField='name'
+                valueShow='cashAccountRef'
+                secondValueShow='cashAccountName'
+                form={formik}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('cashAccountId', newValue ? newValue.recordId : 0)
+                  formik.setFieldValue('cashAccountRef', newValue ? newValue.reference : '')
+                  formik.setFieldValue('cashAccountName', newValue ? newValue.name : '')
+                }}
+                errorCheck={'cashAccountId'}
```
Comment (omarZrayka23) [inline src/pages/ca-rebuild-aba/form/CARebuildForm.js:107]: u do need this

**Review by omarZrayka23 — CHANGES_REQUESTED** [review]

File: src/pages/ca-rebuild-aba/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    datasetId: ResourceIds.CARebuildAccountBalance
+  })
+  const [rendered, setRendered] = useState(false)
+
+  useEffect(() => {
+    if (_labels.accountRebuild && !rendered) {
+      openForm()
+      setRendered(true)
+    }
+  }, [_labels, rendered])
```
Comment (hadichahine) [inline src/pages/ca-rebuild-aba/index.js:20]: Do we need this? Just do _labels

File: src/pages/ca-rebuild-aba/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const CaRebuildAba = () => {
+  const { stack } = useWindow()
+
+  const { labels: _labels, access } = useResourceQuery({
+    datasetId: ResourceIds.CARebuildAccountBalance
+  })
+  const [rendered, setRendered] = useState(false)
+
+  useEffect(() => {
+    if (_labels.accountRebuild && !rendered) {
```
Comment (hadichahine) [inline src/pages/ca-rebuild-aba/index.js:16]: Remove !rendered

File: src/pages/ca-rebuild-aba/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+import { useWindow } from 'src/windows'
+import CARebuildAccountBalance from './form/CARebuildForm'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+
+const CaRebuildAba = () => {
+  const { stack } = useWindow()
+
+  const { labels: _labels, access } = useResourceQuery({
```
Comment (hadichahine) [inline src/pages/ca-rebuild-aba/index.js:10]: try using useResourceParams

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**